tests/setup_test.go

```diff
@@ -2401,3 +2426,301 @@ func TestUpdateOrganization(t *testing.T) {
 		t.Fatalf("org mismatch (-want +got):\n%s", diff)
 	}
 }
+
+func TestOrgInvitation(t *testing.T) {
+	dir := t.TempDir()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	_, c := setup(ctx, t, dir, withOrganizationMemberAddingMode(config.OrganizationMemberAddingModeInvitation))
+
+	gwAdminClient := gwclient.NewClient(c.Gateway.APIExposedURL, c.Gateway.AdminToken)
+
+	//create user01 and user02
+	_, _, err := gwAdminClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: agolaUser01})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	tokenUser01, _, err := gwAdminClient.CreateUserToken(ctx, agolaUser01, &gwapitypes.CreateUserTokenRequest{TokenName: "test"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	gwClientUser01 := gwclient.NewClient(c.Gateway.APIExposedURL, tokenUser01.Token)
+
+	_, _, err = gwAdminClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: "user02"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	tokenUser02, _, err := gwAdminClient.CreateUserToken(ctx, "user02", &gwapitypes.CreateUserTokenRequest{TokenName: "test"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	gwClientUser02 := gwclient.NewClient(c.Gateway.APIExposedURL, tokenUser02.Token)
+
+	//create org01
+	_, _, err = gwClientUser01.CreateOrg(ctx, &gwapitypes.CreateOrgRequest{Name: "org01", Visibility: gwapitypes.VisibilityPublic})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	//create invitation
```

Comment:
You should also check (I can't find it) that directly adding a member as org owner but not agola admin should fail with org member addding mode set to invitation.